src/java.base/share/classes/sun/net/www/protocol/https/HttpsClient.java

```diff
@@ -565,6 +566,7 @@ public void afterConnect() throws IOException, UnknownHostException {
                     // will do the spoof checks in SSLSocket.
                     SSLParameters paramaters = s.getSSLParameters();
                     paramaters.setEndpointIdentificationAlgorithm("HTTPS");
+                    paramaters.setServerNames(Collections.singletonList(new SNIHostName(host)));
```

Comment:
What if the SSL Layer has already an SNIHostName configured? Is there a risk that this will introduce regression in such cases?

Reply:
May be i am not sure, we may need this code change to review by security expert. I am setting "SNIHostName" only if "isDefaultHostnameVerifier" is true(If the HNV is the default from HttpsURLConnection) so there should not be problem.

Reply:
Did you try to run the httpclient tests? They make use of the httpserver - so they can also serve to test it - somewhat.
More generally - please run jdk_net/tier2 tests.

Reply:
i ran tests, it looks clean(https://github.com/vyommani/jdk/actions/runs/612949667)

Reply:
As far as I know this is only tier1 - so none of the network tests have been run.

Reply:
I ran tier1 & tier2 tests locally on my linux vm, it was clear. Can you please do let me know how to run it on OpenJDK build system.